tools/libaqua/libvirt/__init__.py

```diff
@@ -0,0 +1,347 @@
+
+    @classmethod
+    def create(
+        cls,
+        name: str,
+        num_nodes: int,
+        num_disks: int,
+        num_nics: int,
+        deployments_path: Path,
+        mount_path: Optional[Path] = None,
+        uri: Optional[str] = None,
+        cdrom: Optional[str] = None,
+    ) -> Deployment:
+        path = cls._create(name=name, deployments_path=deployments_path)
+        iso_path = None
+        if cdrom:
+            iso_path = str(Path(cdrom).absolute())
+        meta = LibvirtDeploymentModel(
+            name=name,
+            created_on=dt.now(),
+            num_disks=num_disks,
+            num_nics=num_nics,
+            num_nodes=num_nodes,
+            uri=uri,
+            cdrom=iso_path,
+        )
+        deployment = LibvirtDeployment(path, meta)
+        deployment.save()
+
+
+        deployment._create_libvirt_domains()
+
+        return deployment
+
+    def _network_name(self) -> str:
+        return f'{self._meta.name}_network'
+
+    def _node_name(self, index: int) -> str:
+        return f'node{index}'
+
+    def _node_vnc_port(self, index: int) -> int:
+        return 5900 + index
+
+    def node_vnc_port(self, domain_name: str) -> int:
+        node_port = {
+                self._libvirt_domain_name(self._node_name(_)):
+                self._node_vnc_port(_)
+                for _ in range(self._meta.num_nodes)}
+        return node_port[domain_name]
+
+    def _libvirt_domain_name(self, node_name: str) -> str:
+        return f'{self._meta.name}_{node_name}'
+
+    def _libvirt_node_name(self, index: int) -> str:
+        return f'{self._meta.name}_{self._node_name(index)}'
+
+
+    def domain_names(self) -> List[str]:
+        names = [self._libvirt_domain_name(self._node_name(_))
+                            for _ in range(self._meta.num_nodes)]
+        return names
+
+    @staticmethod
+    def numdev(num: int) -> str:
+        return str(chr(ord('a')+num))
+
+    def _disk_source(num: int, pool: str, domain: str) -> str:
+        pass
+
+    def get_disks(self, domain_name: str, pool_name: str) -> Dict:
+        if self._disks is None:
+            self._disks = dict()
+        d = self._disks.get(domain_name, {})
+        if d:
+            return d
+        d = [
+            dict(
+                name=f'{domain_name}_drive{_}',
+                dev=f'vd{self.numdev(_)}',
+                alias=f'virtio-disk{_}',
+                disk_format=self._disk_format,
+                source=f'/mnt/terra/libvirt/images/{pool_name}/{domain_name}_drive{_}.img',
+                    )
+                    for _ in range(self._meta.num_disks)]
+        self._disks[domain_name] = d
+        return d
+
+    def _disks_xml_path(self, name):
+        return self._path.joinpath(f'libivirt_volume_{name}.xml')
+
+
+    @property
+    def _disk_format(self):
+        return 'qcow2'
+
+    def _disk_xml_text(self, name, dev, alias, disk_format, source):
+        disk_size_gb = 8
```

Comment:
lets make this 10, because I'll be changing this value for Vagrant as well soon, and better to just add a nice value to it now rather than later.